src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -352,10 +354,11 @@ REDHAWK_PALEXPORT CONTEXT* PalAllocateCompleteOSContext(_Out_ uint8_t** contextB
     }
 
 #ifdef TARGET_X86
-    if (pfnRtlRestoreContext == NULL)
+    if (pfnRtlRestoreContext == NULL && pfnNtContinue == NULL)
```

Comment:
Should it go in a separate `if` block to match line 444 flow?

Reply:
The assumption is that this `if` block will run exactly once and one of the methods will be found. (I will add an assert.)
It doesn't make sense to split it since both cases would still need to load ntdll and do the lookup.